app/javascript/App/bundles/common/components/NavUser.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react'
+import {
+  Avatar,
+  Divider,
+  IconButton,
+  Menu,
+  MenuItem,
+  ListSubheader,
+} from '@material-ui/core'
+import {
+  AccountCircle,
+  KeyboardArrowDown,
+  KeyboardArrowUp,
+} from '@material-ui/icons'
+import { MenuProps } from '@material-ui/core/Menu'
+import { createStyles, withStyles, withTheme } from '@material-ui/core'
+import { PROFILE_EDIT_URL, LOGOUT_URL } from '~/constants'
+
+const styles = (theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    avatar: {
+      background: 'none',
+    },
+    button: {
+      color: 'white',
+    },
+    menuPaper: {
+      borderRadius: 0,
+      backgroundColor: '#071d29',
+    },
+    menuHeadingItem: {
+      color: 'white',
+      lineHeight: '24px',
+      paddingLeft: 12,
+      paddingRight: 48,
+    },
+    menuAccountLabel: {},
+    menuEmail: {
+      color: 'rgba(255, 255, 255, .7)',
+    },
+    menuList: {
+      paddingTop: 0,
+      paddingBottom: 0,
+    },
+    menuItem: {
+      height: 10,
+      lineHeight: '10px',
+      color: 'white',
+      fontSize: 14,
+      fontWeight: 500,
+      paddingRight: 48,
+    },
+    menuDivider: {
+      backgroundColor: 'rgba(255, 255, 255, .3)',
+    },
+  })
+
+interface Props {
+  menuOpen: boolean
+  menuAnchor: MenuProps['anchorEl']
+  onOpenMenu: (event) => void
+  onCloseMenu: (event) => void
+  formAuthenticityToken: string
+  userEmail: string
+  classes: any
+}
+
+const NavUser: React.StatelessComponent<Props> = (props) => {
+  const {
+    menuOpen,
+    menuAnchor,
+    onOpenMenu,
+    onCloseMenu,
+    formAuthenticityToken,
+    userEmail,
+    classes,
+  } = props
+
+  const LogoutButton = (logoutButtonProps) => (
+    <form method="post" action={LOGOUT_URL}>
+      <input type="hidden" name="_method" value="delete" />
+      <button type="submit" value="Logout" {...logoutButtonProps} />
+      <input
+        type="hidden"
+        name="authenticity_token"
+        value={formAuthenticityToken}
+      />
+    </form>
+  )
+
+  return (
+    <div className={classes.root}>
+      <Avatar className={classes.avatar}>
+        <AccountCircle />
+      </Avatar>
+
+      {menuOpen ? (
+        <IconButton
```

Comment:
Would be nice to style the `IconButton` to be a bit less wide, since the spacing looks a bit weird right now (i.e. the chevron is kinda far from the user profile icon) - not a huge deal though!